### DISCUSSION TITLE
Residents are thirsty--breaking water blocks

### DISCUSSION THREAD

**Discussion by castironduck** [TownyAdvanced/Towny#5212] [Q&A]
I'm relatively new to Towny. It's set up on a small public MC server I'm working on. I made a town for new residents, and guess what, not being able to break blocks means that people have to leave the town to get water. 

As far as I can tell, the only way around this is to configure Towny so that players can break water blocks in farm plots. The downside is that a griefer could just remove all the water, causing hoed farmland to revert to plain dirt. The plan to have the bartender give out water buckets was nixed on the grounds that it creates a source of free water buckets that can then be sold for money.

Is there a more targeted way of dealing with this? Is it possible to create a plot type that allows water blocks to be broken and make a wishing well block?

**Comment by LlmDl** [comment]
Interesting find. Filling a water bucket is considered a destroy test. This might have to be turned into a suggestion some day.

**Reply by castironduck** [reply]
We may have our NPC bartender swap empty buckets for ones full of water. It's a cute solution but not one available to people without an enthusiastic coder on their team.

**Reply by castironduck** [reply]
My coder threw the following snippet together. He said I could go ahead and share it, but didn't want me to share his name given that it's rather quick and dirty. The code intercepts the event and fills the bucket without breaking the water block. No more thirsty players! As you can see, it hard-codes our town name. Also, it applies to the whole town rather than being settable by plot.

[https://pastebin.com/2x13uJcr](https://pastebin.com/2x13uJcr)
